src/services/user/registration.js

```diff
@@ -152,6 +154,13 @@ const registerUser = function register(data, params, app) {
 		)
 		.then(() => {
 			const consentSkipCondition = Configuration.get('SKIP_CONDITIONS_CONSENT');
+			if (
+				(user.roles || []).filter((role) => {
+					return permissionsAllowedToLogin.includes(role);
+				}).length === 0
+			) {
+				return Promise.reject(new errors.BadRequest('You are not allowed to register!'));
```

Comment:
user.roles=['superhero', 'student'] can pass this condition please add this to test, but should not.

const isInvalidRole = (roleNames) => roleNames.some((name) => !permissionsAllowedToLogin.includes(name));

if(isInvalidRole(user.roles)) { }

Reply:
Such code will fail for users which have for example many roles and only one of them is not able to login. Is it a desirable behaviour?

Reply:
I am confuse, the condition should prevent that only roles inpermissionsAllowedToLogin can use to create accounts right?
But if i send ['superhero', 'student'] as roles, i can pass. Because student is valid, but it should not pass because superhero is not valid.

Reply:
const permissionsAllowedToLogin = ['student'];
1) console.log(['student'].filter((role) => permissionsAllowedToLogin.includes(role)).length === 0); 
give me false
2) console.log(['superhero', 'student'].filter((role) => permissionsAllowedToLogin.includes(role)).length === 0); 
give me false too
3) console.log([ 'superhero'].filter((role) => permissionsAllowedToLogin.includes(role)).length === 0); 
give me true

Case 1 and 3 work fine but 2 should run into error.
And if you rewrite the condition it will be fine if you can add a test for it. Where superhero is on array index 0 and a valid on index 1.

Reply:
Yes that is the whole question what if user has for example two roles ,one which allows him to login and one that is forbidden to do that. I assume that if such situation occurs user should not be able to login?

Reply:
Your changes are in registration process not in login.

Reply:
Oh, ok its is registration, then instead login it is still the same question but in terms of registration :D

Reply:
I understand both sides. IMO, @konstantin-sc should decide on that.

Reply:
thou shall not pass! 🧙‍♂️

Reply:
- super heros shouldn't be allowed to use normal sc
- parents are not allowed, too. conflicts like the: the teacher is also a parent of a kid will resolve with our "downgrsading parents"-story (SC-6586), right?

Reply:
Ok, so solution in this ticket should be like: "When user which tries to register have 'superhero' role -> dont allow to do that. In other cases if user has at least one valid role -> can register"? @konstantin-sc

Reply:
For login the superhero should authenticate to server to get the jwt, that is needed for shd operations. But the login to client for shd in blocked on client side. For my understanding the change should only block roles for registration that go over the ui way. But i has not a complete overview over the last tasks.